### ISSUE TITLE
[Bug] Filter with pagination

### ISSUE THREAD

**Issue by vukhanhtruong** [njleonzhang/vue-data-tables#161]
# Online reproduce
https://jsfiddle.net/api/post/library/pure/

## Context

1. Go to page number 2, copy the first column value named "FW201601010002"
2. Paste into the search box
3. Return **No Data**
4. Clean up the search box
5. The pagination shows the current page is **number 1**, but actually is page number 2.
6. Click on the page number 2, the dataset no change

Thanks for your awesome work.

**Comment by njleonzhang** [comment]
tnx for reporting this issue, I will look into later.

**Comment by vukhanhtruong** [comment]
You're welcome. Please let me know once this issue was fixed. Thank you
very much.

**Comment by njleonzhang** [comment]
@vukhanhtruong can you reproduce this issue when use this lib as a `cmd` module instead of a `umd`.

**Comment by njleonzhang** [comment]
The root cause seems related to `element-ui`.
When use `element-ui` as `umd`. The `el-pagination` doesn't update the `internalCurrentPage` when `total` change to 0.

test code:

```
// index.html
<html>

<head>
<link rel="stylesheet" href="https://unpkg.com/element-ui/lib/theme-chalk/index.css">
<script src="http://unpkg.com/vue/dist/vue.js"></script>
<script src="http://unpkg.com/element-ui/lib/index.js"></script>
<script src="http://unpkg.com/element-ui/lib/umd/locale/en.js"></script>
<script src="http://unpkg.com/vue-data-tables/dist/data-tables.js"></script>
</head>

<body>
  <div id="app">
    <data-tables :data="data" :actions-def="actionsDef" :checkbox-filter-def="checkFilterDef" :action-col-def="actionColDef">
      <el-table-column v-for="title in titles" :prop="title.prop" :label="title.label" sortable="custom">
      </el-table-column>
    </data-tables>
  </div>

  <script>
    ELEMENT.locale(ELEMENT.lang.en)
    Vue.use(DataTables)
    Vue.use(DataTables.DataTablesServer)
    var data, titles

    data = [...new Array(100)].map((_, index) => {
      return {
      "content": "Water flood",
      "flow_no": `FW20160101000${index}`,
      "flow_type": "Repair",
      "flow_type_code": "repair",
      }
    })

    titles = [{
      prop: "flow_no",
      label: "NO."
      }, {
      prop: "content",
      label: "Content"
      }, {
      prop: "flow_type",
      label: "Type"
    }]

    var Main = {
      data() {
        return {
          data,
          titles,
          actionsDef: {
              colProps: {
                span: 5
              },
              def: [{
                name: 'new',
                handler: () => {
                  this.data.push({
                    'content': 'hello world',
                    'flow_no': 'FW201601010004',
                    'flow_type': 'Help',
                    'flow_type_code': 'help',
                  })
                },
                buttonProps: {
                  type: 'text'
                }
              }, {
                name: 'import',
                handler: () => {
                  this.$message('import clicked')
                },
                icon: 'upload'
              }]
            },
            checkFilterDef: {
              props: 'flow_type_code',
              def: [{
                'code': 'repair',
                'name': 'Repair'
              }, {
                'code': 'help',
                'name': 'Help'
              }]
           },
           actionColDef: {
               label: 'Actions',
              def: [{
                handler: row => {
                  this.$message('Edit clicked')
                  row.flow_no = "hello word"
                },
                name: 'Edit'
              }, {
                icon: 'message',
                type: 'text',
                handler: row => {
                  this.$message('RUA in row clicked')
                  console.log('RUA in row clicked', row)
                },
                name: 'RUA'
              }]
            }
        }
      }
    }
    var Ctor = Vue.extend(Main)
    var vue = new Ctor().$mount('#app')
  </script>
</body>
</html>
```
open this file in browser, and then do the following:

1. in search box, type in "leon"
2. Return **No Data**

in this time, the filtered data should be empty, just as you can see `no data`, and the `el-pagination` get the total as 0. now the `internalCurrentPage` property of `el-pagination` should change to 1, and `current-change` should be emitted with value 1. But issue occurs here.  `internalCurrentPage` is changed to 1, but `current-change` is not emitted, which cause that the `currentPage` of `vue-data-tables` is not notified to update.

![88ff4fcc-29a8-4abb-adf4-2582aa0a1f4b](https://user-images.githubusercontent.com/13174059/38488096-e5a34830-3c14-11e8-895b-08b6eb108f89.png)

Now this issue only happened when use element-ui in `umd` mode, if you use `element-ui` and `vue-data-talbes` as module in `webpack` or `rollup`, it seems this issue doesn't occur.

This issue is wired, it may take more time to investigate and report to `element-ui` team. I suggest you try to use this lib in webpack and try if you can workaround this issue.

**Comment by njleonzhang** [comment]
confirmed. It should be a element-ui bug and can reproduce without vue-data-tables.
I have submitted a bug: https://github.com/ElemeFE/element/issues/10625

**Comment by vukhanhtruong** [comment]
I really appreciate for your support. Thank you very much. Keep your good
work up :)

**Comment by njleonzhang** [comment]
element-ui 2.3 change the behavior, which cause this issue. I has fixed this issue in `vue-data-tables` v3.2.0, you can have a try with new version.
